src/containers/forms/AssignTeamTableCell/index.tsx

```diff
@@ -0,0 +1,145 @@
+
+reducerRegistry.register(assignmentReducerName, assignmentReducer);
+reducerRegistry.register(organizationsReducerName, organizationsReducer);
+
+/** Interface for Assign Teams cell props */
+export interface AssignTeamCellProps {
+  assignments: Assignment[]; // all the assignments for the plan-jurisdiction
+  assignmentsArray: Assignment[]; // all the assignments for the plan
+  assignButton?: React.ElementType;
+  assignPopover?: React.ElementType;
+  resetPlanAssignmentsAction: typeof resetPlanAssignments;
+  jurisdictionId: string;
+  organizationsById: { [key: string]: Organization } | null;
+  planId: string;
+}
+
+/** Component that will be rendered in IRS planning table cells
+ * showing the button and card in the Assign Teams column
+ */
+const AssignTeamTableCell = (props: AssignTeamCellProps) => {
+  const {
+    assignments,
+    assignmentsArray,
+    assignButton,
+    assignPopover,
+    resetPlanAssignmentsAction,
+    jurisdictionId,
+    organizationsById,
+    planId,
+  } = props;
+  const [isActive, setIsActive] = useState<boolean>(false);
+
+  const onPlanAssignmentButtonClick = (e: MouseEvent) => {
+    stopPropagationAndPreventDefault(e);
+    setIsActive(!isActive);
+  };
+
+  const onClearAssignmentsButtonClick = (e: MouseEvent) => {
+    stopPropagationAndPreventDefault(e);
+    const nextAssignments = assignmentsArray.filter(
+      (a: Assignment) => a.jurisdiction !== jurisdictionId
+    );
+    resetPlanAssignmentsAction({ [planId]: [...nextAssignments] });
+    setIsActive(!isActive);
+  };
+
+  const onSaveAssignmentsButtonClick = (e: MouseEvent) => {
+    stopPropagationAndPreventDefault(e);
+    setIsActive(!isActive);
+  };
+
+  const AssignTeamButton = (
+    <Button
+      color="primary"
+      className="assign-team-button"
+      id={getButtonId(jurisdictionId)}
+      onClick={onPlanAssignmentButtonClick}
+      size="sm"
+      style={{ float: 'right' }}
+    >
+      {ASSIGN_TEAMS}
+    </Button>
+  );
+
+  const defaultAssignTeamPopoverProps: AssignTeamPopoverProps = {
+    formName: getFormName(jurisdictionId),
+    isActive,
+    jurisdictionId,
+    onClearAssignmentsButtonClick,
+    onSaveAssignmentsButtonClick,
+    onToggle: () => setIsActive(!isActive),
+    organizationsById,
+    planId,
+    target: getButtonId(jurisdictionId),
+  };
+  const defaultAssignTeamPopover = <AssignTeamPopover {...defaultAssignTeamPopoverProps} />;
+
+  return (
+    <div onClick={stopPropagationAndPreventDefault}>
+      <span style={{ paddingRight: '2rem' }} className="assignment-count-text">
+        {`${assignments.length} ${TEAMS_ASSIGNED}`}
+      </span>
+      {assignButton || AssignTeamButton}
+      {assignPopover || defaultAssignTeamPopover}
+    </div>
+  );
+};
+
+const getButtonId = (jurisdictionId: string): string => `plan-assignment-${jurisdictionId}`;
+const getFormName = (jurisdictionId: string): string => `plan-assignment-form-${jurisdictionId}`;
```

Comment:
I feel like these would be happier/cleaner if they came from a helpers/utils file

Reply:
moved